scripts/docs/build-features-overview.js

```diff
@@ -0,0 +1,456 @@
+		// Add new line after `<tag>` or `</tag>`, but only if it is not already followed by a new line (negative lookahead).
+		.replace( /<(\/)?(table|thead|tbody|tr|th|td|p|a)(.*?)>(?!\n)/g, '<$1$2$3>\n' )
+		// Remove whitespace before `>`, that may appear there after adding an empty attribute.
+		.replace( /\s+>/g, '>' )
+		// Divide input string into lines, which start with either an opening tag, a closing tag, or just a text.
+		.split( '\n' );
+
+	let indentCount = 0;
+
+	return lines
+		.map( ( line, index ) => {
+			if ( isOpeningTag( line ) && hasClosingTagFor( line, lines.slice( index ) ) ) {
+				return indentLine( line, indentCount++ );
+			}
+
+			if ( isClosingTag( line ) ) {
+				return indentLine( line, --indentCount );
+			}
+
+			return indentLine( line, indentCount );
+		} )
+		.join( '\n' );
+}
+
+/**
+ * Checks, if an argument is an opening tag.
+ *
+ * @param {String} line String to check.
+ * @returns {Boolean}
+ */
+function isOpeningTag( line ) {
+	return line.startsWith( '<' ) && !isClosingTag( line );
+}
+
+/**
+ * Checks, if an argument is a closing tag.
+ *
+ * @param {String} line String to check.
+ * @returns {Boolean}
+ */
+function isClosingTag( line ) {
+	return line.startsWith( '</' );
+}
+
+/**
+ * Checks, if there is a closing tag for currently examined opening tag.
+ *
+ * @param {String} tag Currently examined opening tag.
+ * @param {Array.<String>} lines Next lines to search for a closing tag.
+ * @returns {Boolean}
+ */
+function hasClosingTagFor( tag, lines ) {
+	const closingTag = tag.replace( /<([a-z]+).*>/, '</$1>' );
+
+	return lines.some( line => line === closingTag );
+}
+
+/**
+ * Indents a line by a specified number of characters.
+ *
+ * @param {String} line Line to indent.
+ * @param {Number} indentCount Number of characters to use for indentation.
+ * @param {String} [indentChar] Indentation character.
+ * @returns {String}
+ */
+function indentLine( line, indentCount, indentChar = '\t' ) {
+	return `${ indentChar.repeat( indentCount ) }${ line }`;
+}
+
+/**
+ * Helper, which transforms any value to an array. If the provided value is already an array, it is returned unchanged.
+ *
+ * @param {*} data The value to transform to an array.
+ * @returns {Array.<*>} An array created from data.
+ */
+function toArray( data ) {
+	return Array.isArray( data ) ? data : [ data ];
+}
+
+/**
+ * Helper (factory), which creates a function, that prepends and/or appends provided value by another value.
+ *
+ * @param {Object} options Options to define prefix and/or suffix.
+ * @param {String} [options.prefix] A string to add as a prefix to provided value. Empty string by default.
+ * @param {String} [options.suffix] A string to add as a suffix to provided value. Empty string by default.
+ * @returns {Function}
+ */
+function wrapBy( { prefix = '', suffix = '' } = {} ) {
+	return item => `${ prefix }${ item }${ suffix }`;
+}
+
+/**
+ * @typedef {Object.<String, String|Array.<String>>} HtmlOutput
```

Comment:
I think that you can specify the base type to just `Object` in all these typedefs. `Object.<String, String>` and similar are a bit misleading as they suggest that the object behaves like a key-value dictionary, while it only accepts known properties.

Reply:
Good idea 👍🏻